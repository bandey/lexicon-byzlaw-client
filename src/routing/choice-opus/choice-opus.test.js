import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

import enzymeToJson from 'enzyme-to-json';

import {BrowserRouter} from 'react-router-dom';
import DataProvider from '../../data-provider/data-provider.js';
import Choice from '../../components/choice/choice.js';

import ChoiceOpus from './choice-opus.js';

const data = [
  {id: 'one', name: 'Alpha'},
  {id: 'two', name: 'Betta'},
  {id: 'three', name: 'Gamma'},
];

// Module.mock must be in the same scope as import Module
jest.mock('../../data-provider/data-provider.js');
DataProvider.mockImplementation(({children, query}) => {
  const Children = children;
  return (<Children data={data} />);
});

jest.mock('../../components/choice/choice.js');
Choice.mockImplementation(({title, children, wrapLink}) => {
  return (
    <React.Fragment>
      {children.map(item => wrapLink(item.id, item.name))}
    </React.Fragment>
  );
});

it('uses correct query', () => {
  const rendered = shallow(<ChoiceOpus />);

  // console.log(rendered.debug());

  const provider = rendered.find('DataProvider');
  expect(provider.exists()).toBeTruthy();
  expect(provider.prop('query')).toEqual({
    name: 'opusAll',
    gql: 'opusAll { id name }',
  });
});

it('renders correct content', () => {
  const rendered = mount(
    <BrowserRouter>
      <ChoiceOpus />
    </BrowserRouter>
  );

  // console.log(rendered.debug());

  expect(enzymeToJson(rendered)).toMatchSnapshot();
});
