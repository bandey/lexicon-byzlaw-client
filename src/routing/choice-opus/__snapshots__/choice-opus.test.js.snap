// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct content 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <ChoiceOpus
      match={
        Object {
          "url": "/route/alone",
        }
      }
      title="Choose"
    >
      <DataProvider
        query={
          Object {
            "gql": "opusAll { id name }",
            "name": "opusAll",
          }
        }
      >
        <AdapterChoice
          data={
            Array [
              Object {
                "id": "one",
                "name": "Alpha",
              },
              Object {
                "id": "two",
                "name": "Betta",
              },
              Object {
                "id": "three",
                "name": "Gamma",
              },
            ]
          }
        >
          <ChoiceWithFontPromo
            title="Choose"
            wrapLink={[Function]}
          >
            <Link
              key="one"
              onClick={null}
              to="/route/alone/one"
            >
              <LinkAnchor
                href="/route/alone/one"
                navigate={[Function]}
                onClick={null}
              >
                <a
                  href="/route/alone/one"
                  onClick={[Function]}
                >
                  Alpha
                </a>
              </LinkAnchor>
            </Link>
            <Link
              key="two"
              onClick={null}
              to="/route/alone/two"
            >
              <LinkAnchor
                href="/route/alone/two"
                navigate={[Function]}
                onClick={null}
              >
                <a
                  href="/route/alone/two"
                  onClick={[Function]}
                >
                  Betta
                </a>
              </LinkAnchor>
            </Link>
            <Link
              key="three"
              onClick={null}
              to="/route/alone/three"
            >
              <LinkAnchor
                href="/route/alone/three"
                navigate={[Function]}
                onClick={null}
              >
                <a
                  href="/route/alone/three"
                  onClick={[Function]}
                >
                  Gamma
                </a>
              </LinkAnchor>
            </Link>
          </ChoiceWithFontPromo>
        </AdapterChoice>
      </DataProvider>
    </ChoiceOpus>
  </Router>
</BrowserRouter>
`;
